python/bindings.cpp

```diff
@@ -316,12 +320,9 @@ void bindReportReader(py::module& m, const std::string& prefix) {
                                    (prefix + "DataFrame").c_str(),
                                    "Something easily convertible to pandas dataframe")
         .def_readonly("ids", &DataFrame<KeyType>::ids)
-        .def_property_readonly("data",
-                               [](DataFrame<KeyType>& dframe) {
-                                   return asArray(std::move(dframe.data),
-                                                  dframe.n_cols,
-                                                  dframe.n_rows);
-                               })
+        .def_property_readonly("data", [](DataFrame<KeyType>& dframe) {
```

Comment:
Stupid question… is the `data` property cached? Otherwise, due to the `std::move`, this would be a one-shot operation.

Reply:
that's a very good point. `asArray` is typically used right after we extracted the data from some place, to avoid making copies of temp data. Indeed it seems the source struct is not temp here... so the user can't call it twice. 
Maybe the data is too large to make copies and we should attempt to share/cache the array?